### PR TITLE
File Dialog

### PR DESCRIPTION
**Pull request by Altenhh**

### CODE REVIEW

File: GDEdit/GDE.App/Main/UI/FileDialogComponents/FileDialog.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            });
+
+            filePathBreadcrumbs.Items.AddRange(testValues);
```
Comment (Altenhh) [inline GDEdit/GDE.App/Main/UI/FileDialogComponents/FileDialog.cs]: This is the code you'll need to write for grabbing all directories/files. if you want to change the icon just do `Icon = FontAwesome.smthin.icon`, this also applies to name and color

File: GDEdit/GDE.App/Main/UI/FileDialogComponents/FileDialog.cs [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+            var splitA = AnalyzePath(pathA);
+            var splitB = AnalyzePath(pathB);
+            var result = new List<string>();
+            int min = Min(splitA.Length, splitB.Length);
+            for (int i = 0; i < min; i++)
+                if (splitA[i] == splitB[i])
+                    result.Add(splitA[i]);
+            return result.Aggregate(AggregateDirectories);
+        }
+        private static string GetPreviousPathDirectoryInNewPath(string previousPath, string newPath)
```
Comment (Altenhh) [inline GDEdit/GDE.App/Main/UI/FileDialogComponents/FileDialog.cs]: These should be camelCase

File: GDEdit/GDE.App/Main/UI/Containers/GDEScrollContainer.cs [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+using osuTK.Graphics;
+using osuTK.Input;
+
+namespace GDE.App.Main.UI.Containers
+{
+    public class GDEScrollContainer : ScrollContainer<Drawable>
+    {
+        private bool mouseScrollBarDragging;
+
+        /// <summary>
```
Comment (Altenhh) [inline GDEdit/GDE.App/Main/UI/Containers/GDEScrollContainer.cs]: i forgot to remove spacing on the summary tags lmao

File: GDEdit/GDE.App/Main/UI/FileDialogComponents/FileDialog.cs (hunk 2 above)
Comment (AlFasGD) [reply GDEdit/GDE.App/Main/UI/FileDialogComponents/FileDialog.cs]: You're actually insane, we've agreed on the exact opposite convention.

**Comment by AlFasGD** [comment]
Known shortcut issues: The PageUp, PageDown, Home, End keys are primarily handled by the scroll container, if focused. This means that to ues these shortcuts in the file container you need to focus out of it entirely.

**Pull request merged**